src/signer.ts

```diff
@@ -11,6 +12,31 @@ const rippleCodec = require("ripple-binary-codec");
  * Abstracts the details of signing.
  */
 class Signer {
+  /**
+   * Encode the given object to hex and sign it.
+   *
+   * @param transaction The transaction to sign.
+   * @param wallet The wallet to sign the transaction with.
+   * @returns A signed transaction.
+   */
+  public static signTransaction(
+    transaction: Transaction,
+    wallet: Wallet
+  ): string | undefined {
+    if (transaction === undefined || wallet === undefined) {
+      return undefined;
+    }
+
+    var transactionJSON = Serializer.transactionToJSON(transaction);
+    if (transactionJSON === undefined) {
+      return undefined;
+    }
+    const transactionHex = rippleCodec.encodeForSigning(transactionJSON);
+
+    const signatureHex = wallet.sign(transactionHex);
+    return signatureHex;
```

Comment:
Do you want the `signatureHex` variable here to name this, rather than just returning it?

Reply:
Nope, just a poorly finished thought after copy / pasting the legacy method below.